### PR TITLE
[MAINTENANCE] `ExpectationSuite` API cleanup

### PR DESCRIPTION
**Pull request by cdkini**
Delete any extraneous methods we don't need (citations, filtering, etc). Also, make any methods private if we can.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/core/test_expectation_suite_crud_methods.py [hunk 1]
```diff
@@ -379,7 +379,14 @@ def test_add_expectation_configurations(
...
 
-    assert single_expectation_suite.isEquivalentTo(different_suite)
+    assert len(single_expectation_suite.expectations) == len(different_suite.expectations)
+    assert all(
+        mine.isEquivalentTo(theirs)
+        for (mine, theirs) in zip(
+            single_expectation_suite.expectation_configurations,
+            different_suite.expectation_configurations,
+        )
+    )
```
Comment (cdkini) [inline tests/core/test_expectation_suite_crud_methods.py:324]: Took the guts of the `isEquivalentTo` method and used it for the assertion here

File: great_expectations/core/expectation_suite.py [hunk 2]
```diff
@@ -507,26 +389,7 @@ def remove_expectation(
...
-        self.expectations = [
-            expectation
-            for expectation in self.expectations
-            if expectation.expectation_type not in expectation_types
-        ]
-
-        return removed_expectations
-
-    def find_expectation_indexes(
+    def _find_expectation_indexes(
```
Comment (cdkini) [inline great_expectations/core/expectation_suite.py:392]: This is used by other methods so it has to stay around but I don't think it needs to be public

File: great_expectations/core/expectation_suite.py [hunk 3]
```diff
@@ -273,84 +264,6 @@ def expectation_configurations(self, value):
...
-            "batch_markers": batch_markers,
-            "batch_parameters": batch_parameters,
-            "profiler_config": profiler_config,
-            "comment": comment,
-        }
-        gx.util.filter_properties_dict(properties=citation, clean_falsy=True, inplace=True)
-        self.meta["citations"].append(citation)
-
-    # noinspection PyPep8Naming
-    def isEquivalentTo(self, other):
```
Comment (tyler-hoffman) [inline great_expectations/core/expectation_suite.py:320]: I thought this one was too hard to pull out? Awesome if we can do it now!

File: great_expectations/core/expectation_suite.py (hunk 2 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_suite.py:392]: I feel like we're misusing this. This method isn't private; it's deprecated. And it's already not in the public api. `typing-extensions` has a `deprecated` decorator. Thoughts on just using that?

**Review by tyler-hoffman — CHANGES_REQUESTED** [review]
I want to at least discuss not using the psuedo-private `_` prefix for methods that are used outside of the class.

File: great_expectations/core/expectation_suite.py (hunk 2 above)
Comment (tyler-hoffman) [reply great_expectations/core/expectation_suite.py:392]: I suppose I care less about the distinction if the only external use case is in a test

**Review by tyler-hoffman — APPROVED** [review]

File: great_expectations/core/expectation_suite.py (hunk 2 above)
Comment (cdkini) [reply great_expectations/core/expectation_suite.py:392]: Just to close the loop on this, this method is only ever used within this class (at least after the other deletions in this file).

**Pull request merged**